src/darker/__main__.py

```diff
@@ -79,7 +79,16 @@ def format_edited_parts(
         else:
             edited = worktree_content
         max_context_lines = len(edited.lines)
-        for context_lines in range(max_context_lines + 1):
+        minimum_context_lines = BinarySearch(0, max_context_lines + 1)
+        last_successful_reformat = None
+        while not minimum_context_lines.found:
+            context_lines = minimum_context_lines.get_next()
+            if context_lines > 0:
+                logger.warning(
```

Comment:
Was this changed to a warning for debugging purposes?
Looks like `debug` was the appropriate level